app/controllers/start_group_controller.rb

```diff
@@ -0,0 +1,51 @@
+class StartGroupController < ApplicationController
+  def new
+    @errors = []
+    @group = Group.new
+  end
+
+  def create
+    if user_signed_in?
+      create_signed_in
+    else
+      create_signed_out
+    end
+  end
+
+  def create_signed_in
+    @group = Group.new(permitted_params.group)
+    @errors = []
+    @errors << 'is_commercial' if @group.is_commercial.nil?
+    @errors << 'group_name' if @group.name.blank?
+
+    # check for valid name and email
+    if @group.valid? and @errors.empty?
+      StartGroupService.start_group(@group)
+      @group.add_admin!(current_user)
+      redirect_to @group
+    else
+      render :new
+    end
+  end
+
+  def create_signed_out
+    @group = Group.new(permitted_params.group)
+    @email = params[:email]
+    @name =  params[:name]
+    @errors = []
+    @errors << 'name' if @name.blank?
+    @errors << 'email' unless /^[^@]+@[^@]+\.[^@]+$/.match @email
```

Comment:
These should happen on the Invitation model, I reckon. I'd feel a lot better if this controller method looked more like

```
@group = Group.new(permitted_params.group)
@invitation = Invitation.new(name: params[:name], email: params[:email]) # xtra credit for permitted_params.invitation
if @group.valid? && @invitation.valid? # perform email / name validations here
  StartGroupService.start_group(group: @group, admin_invitation: @invitation)
else
  @errors = {}.extend @group.errors, @invitation.errors
end
```